YogaKit/YKLayout.h

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "YKEnums.h"
+
+@interface YKLayout : NSObject
+
+/**
+ The property that decides if we should include this view when calculating layout. Defaults to YES.
+ */
+@property (nonatomic, setter=setIncluded:) BOOL isIncluded;
+
+/**
+ The property that decides during layout/sizing whether or not yk_* properties should be applied. Defaults to NO.
+ */
+@property (nonatomic, setter=setEnabled:) BOOL isEnabled;
+
+@property (nonatomic) YKDirection direction;
+@property (nonatomic) YKFlexDirection flexDirection;
+@property (nonatomic) YKJustify justifyContent;
+@property (nonatomic) YKAlign alignContent;
+@property (nonatomic) YKAlign alignItems;
+@property (nonatomic) YKAlign alignSelf;
+@property (nonatomic) YKPositionType positionType;
+@property (nonatomic) YKWrap flexWrap;
+
+@property (nonatomic) CGFloat flexGrow;
+@property (nonatomic) CGFloat flexShrink;
+@property (nonatomic) CGFloat flexBasis;
+
+@property (nonatomic) CGFloat positionLeft;
+@property (nonatomic) CGFloat positionTop;
+@property (nonatomic) CGFloat positionRight;
+@property (nonatomic) CGFloat positionBottom;
+@property (nonatomic) CGFloat positionStart;
+@property (nonatomic) CGFloat positionEnd;
+@property (nonatomic) CGFloat positionHorizontal;
```

Comment:
Hey @hartbit, sorry I took so long to respond. I really appreciate the time you put into that response, and I support all of it 100%.

@emilsjolander I know we talked about `YGEdgeValue`, but, I think @hartbit makes a compelling case for why we should put these properties on the layout object. 

1. Matches other UIKit APIs.
2. Less objects on the heap.
3. One less method lookup when we are getting/setting this on the objc runtime.